packages/@aws-cdk/core/lib/private/tree-metadata.ts

```diff
@@ -49,10 +52,27 @@ export class TreeMetadata extends Construct {
       }
     });
   }
+
+  private inferAttributes(construct: IConstruct): { [key: string]: any } | undefined {
```

Comment:
You're not inferring though, you're querying.

Reply:
changed to `getAttributes`... will need to add token support soon :)